### PR TITLE
[chore] [processor/k8sattributes] improve documentation for the k8sattributes processor

### PR DESCRIPTION
**Pull request by bacherfl**
**Description:** This PR attempts to improve the documentation of the k8sattributes processor. Being relatively new to this component I tried to make the documentation a bit easier to understand for new users, and highlight some limitations I found while going through the implementation.

**Link to tracking Issue:** #32104

**Testing:** Trying out different configurations and association rules to get a better understanding

**Documentation:** Extended the readme of this component

### CODE REVIEW

File: processor/k8sattributesprocessor/README.md
Comment (bacherfl) [inline processor/k8sattributesprocessor/README.md:38]: please let me know if this list is incomplete or wrong, but I thought it might be worth having this explicit list to avoid confusion regarding which attributes can be used for these rules (e.g. the limitation of container level attributes not working here)

File: processor/k8sattributesprocessor/README.md
Comment (bacherfl) [inline processor/k8sattributesprocessor/README.md:112]: may be a trivial example, but it's always helpful especially for new users to see exactly where these properties need to be set. Can remove it though if it makes the readme too verbose

File: processor/k8sattributesprocessor/README.md
Comment (ChrsMark) [inline processor/k8sattributesprocessor/README.md:113]: While this example is useful, I think we lack an example showing what the output of the processor looks like given specific inputs.

File: processor/k8sattributesprocessor/README.md
Comment (ChrsMark) [inline processor/k8sattributesprocessor/README.md:45]: Shouldn't these be either enabled by default or explicitly set by the `metadata` rules in order to be available in the association? If so, I think it makes sense to mention that.

**Review by ChrsMark — COMMENTED** [review]
Thank's for working on this!

I also think that the list at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33800/files#diff-57b0d6448d516c0c80b326a7f7f95c3256b2b774e496870dc1dd35dcf8bfc256R101 is a bit hard to interpret and leaves some parts as assumptions? 
Could we also revisit this as well?
I'm particularly interested about if the `container.id` is populated when **only** the `k8s.container.name` is provided. If that's the case, then the documentation needs to be fixed. If not, then I think this should be fixed in the implementation since I guess it should be doable to get the `container.id` as well (at [`extractPodContainersAttributes`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/97f46cc29f0bd7170f0a169efca96d7b010b9024/processor/k8sattributesprocessor/internal/kube/client.go#L632)) (filing a different issue to track this should be fine).

**Comment by bacherfl** [comment]
> Thank's for working on this!
> 
> I also think that the list at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33800/files#diff-57b0d6448d516c0c80b326a7f7f95c3256b2b774e496870dc1dd35dcf8bfc256R101 is a bit hard to interpret and leaves some parts as assumptions? Could we also revisit this as well? I'm particularly interested about if the `container.id` is populated when **only** the `k8s.container.name` is provided. If that's the case, then the documentation needs to be fixed. If not, then I think this should be fixed in the implementation since I guess it should be doable to get the `container.id` as well (at [`extractPodContainersAttributes`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/97f46cc29f0bd7170f0a169efca96d7b010b9024/processor/k8sattributesprocessor/internal/kube/client.go#L632)) (filing a different issue to track this should be fine).

Thanks for your feedback, @ChrsMark! You're right - I was not 100% sure if I should also revisit this section - After trying out some configurations and debugging I verified that those additional container attributes are indeed only added if the `k8s.container.name` is present in the resource attributes, so it is technically correct. But maybe the requirement of having this attribute in the resource attributes should be highlighted more explicitly since this might be overlooked otherwise. I will try to also reword this section a little

File: processor/k8sattributesprocessor/README.md
Comment (bacherfl) [reply processor/k8sattributesprocessor/README.md]: good point, I have moved the list to the section about extending the metadata and added a remark that they need to be set there to be available in the association

File: processor/k8sattributesprocessor/README.md
Comment (bacherfl) [reply processor/k8sattributesprocessor/README.md:118]: you're right - having an example here should make things clearer - I have added a small example to illustrate how the attributes of a span are extended based on the properties of a matching pod

**Comment by bacherfl** [comment]
@ChrsMark when you have some time, can you give this one another review please?

**Review by ChrsMark — COMMENTED** [review]
The added content looks good to me.

My only concern is if we actually covered the following:
> After trying out some configurations and debugging I verified that those additional container attributes are indeed only added if the k8s.container.name is present in the resource attributes, so it is technically correct.

I still think it's a tricky part which either needs to be clearly documented and/or be fixed within the implementation so as to provide the same set of metadata in all cases (if possible).

**Comment by bacherfl** [comment]
> The added content looks good to me.
> 
> My only concern is if we actually covered the following:
> 
> > After trying out some configurations and debugging I verified that those additional container attributes are indeed only added if the k8s.container.name is present in the resource attributes, so it is technically correct.
> 
> I still think it's a tricky part which either needs to be clearly documented and/or be fixed within the implementation so as to provide the same set of metadata in all cases (if possible).

Thank you for the feedback @ChrsMark! When working on this it seemed to me that this particular point was actually already covered in the original readme in this section: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33800/files#diff-57b0d6448d516c0c80b326a7f7f95c3256b2b774e496870dc1dd35dcf8bfc256R97-R106 

I hope the additional example added in this PR helps to make this a bit clearer, but I agree that this is a bit of a confusing requirement to begin with, so having a closer look at the implementation to see if this can be improved might be a good idea (at least for pods running a single container it should be possible to automatically attach container level attributes)

**Comment by bacherfl** [comment]
FYI - I have created an issue to discuss the possibility of falling back to the only container in a single-container pod to relax the requirement of having to provide the `k8s.container.name`/`container.id` attributes here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34189

**Review by ChrsMark — APPROVED** [review]

**Review by evan-bradley — APPROVED** [review]
Looks good to me.

**Comment by evan-bradley** [comment]
@dmitryax @fatsheep9146 @rmfitzpatrick @TylerHelmuth could one of you please take a look?

File: processor/k8sattributesprocessor/README.md
Comment (tiffany76) [inline processor/k8sattributesprocessor/README.md:110]:
```suggestion
Please note, however, that container level attributes can't be used for source rules in the pod_association.
```

File: processor/k8sattributesprocessor/README.md
Comment (tiffany76) [inline processor/k8sattributesprocessor/README.md:128]:
```suggestion
The previous configuration attaches the attributes listed in the `metadata` section to all resources received by a matching pod with the `k8s.container.name` attribute being present. For example, when the following trace
```

File: processor/k8sattributesprocessor/README.md
Comment (tiffany76) [inline processor/k8sattributesprocessor/README.md:173]:
```suggestion
the processor associates the received trace to the pod, based on the connection IP, and add those attributes to the resulting span:
```

**Review by tiffany76 — COMMENTED** [review]
Docs approver here. Just a few small copy edits. Thanks!

**Comment by bacherfl** [comment]
> Docs approver here. Just a few small copy edits. Thanks!

Thanks for the suggestions, @tiffany76! I have included them into the PR

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**
